server/server.py

```diff
@@ -9,20 +12,30 @@ def model():
       app.logger.warn("BODY: %s" % ret)
       return ret
 
+
 @app.route('/data',methods = ['POST', 'GET'])
 def data():
    if request.method == 'POST':
       app.logger.warn("BODY: %s" % request.get_data())
       return 'success'
 
-#Required to send models. There are no models created yet.
+
+# Returns a trained model given the model type/name. Currently, the route will return an archived file of keras's SavedModel.
+# The extracted archived SavedModel can be directly fitted. To download the SavedModel as attachment, include as_attachment=True in the 
+# send_from_directory flask function. Other lightweight options include sending the model as a HDF5 format, sending the model
+# structure as a JSON and the model's weights as an HDF5 file. 
 @app.route('/models/<model_type>', methods=['GET'])
-def get_model(model_type='core'):
-    models_directory = os.path.join(current_app.root_path, '/models/')
-    if model_type == 'dram':
-        return send_from_directory(directory=models_directory, filename='dram_model')
-    elif model_type == 'core':
-        return send_from_directory(directory=models_directory, filename='core_model')
+@app.route('/models/')
+def get_model(model_type='core_model'):
+    models_directory = os.path.join(current_app.root_path, 'models/')
+    if model_type == 'dram_model' and os.path.exists(os.path.join(models_directory, 'dram_model')): # must check whether the model was created or not
+        shutil.make_archive(os.path.join(models_directory, 'dram_model'), 'zip', os.path.join(models_directory, 'dram_model')) # this function should overrite an existing zipped model
+        return send_from_directory(models_directory, 'dram_model.zip')
+    elif model_type == 'core_model' and os.path.exists(os.path.join(models_directory, 'core_model')): # must check whether the model was created or not
```

Comment:
Nit: not sure if this pattern is common in this code (and if it is then ignore my comment), but you don't need `elif` when the code unconditionally `return`s in the `if` above

Reply:
Created new PR removing the verbose code.